pyomo/contrib/cspline_external/test/test_parameter_calculation.py

```diff
@@ -0,0 +1,75 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import os
+
+from pyomo.common.dependencies import attempt_import
+from pyomo.contrib.cspline_external.cspline_parameters import (
+    cubic_parameters_model,
+    get_parameters,
+)
+from pyomo.opt import check_available_solvers
+import pyomo.environ as pyo
+from pyomo.common.fileutils import this_file_dir
+import pyomo.common.unittest as unittest
+
+np, numpy_available = attempt_import("numpy")
+
+
+@unittest.skipUnless(
+    check_available_solvers("ipopt"), "The 'ipopt' solver is not available"
+)
+@unittest.skipIf(not numpy_available, "numpy is not available.")
+class CsplineExternalParamsTest(unittest.TestCase):
+    def test_param_gen(self):
+        x_data = [1, 2, 3, 4, 5]
+        y_data = [2, 3, 5, 2, 1]
+
+        m = cubic_parameters_model(x_data, y_data)
+
+        # This is a linear system of equations by default, so although ipopt
+        # is not needed, it works perfectly well.
+        solver_obj = pyo.SolverFactory("ipopt")
+        solver_obj.solve(m)
+
+        params = os.path.join(this_file_dir(), "test.txt")
```

Comment:
Future-proofing: we should not assume the source directory is writable.  If this needs to be a file, then we should put it in TEMP.

Reply:
Switched to string stream and just totally avoid file write.